tests/functionality_tests/test_activation_shapes.py

```diff
@@ -44,11 +71,22 @@ def main():
         y3 = cols // 6 ## Good value 5
         grid.activate_triangle(x1,y1,x2,y2,x3,y3) 
```

Comment:
When the activation type is triangle, an error is thrown for isInsideTriangle since it's not initialized in the Grid class. 
 if self.isInsideTriangle(x1, y1, x2, y2, x3, y3, x, y):
AttributeError: 'Grid' object has no attribute 'isInsideTriangle'